tests/otel_integrations/test_httpx.py

```diff
@@ -348,3 +348,213 @@ def test_missing_opentelemetry_dependency() -> None:
+        with httpx.Client(transport=create_transport()) as client:
+            logfire.instrument_httpx(
+                client,
+                capture_request_headers=True,
+                capture_request_json_body=True,
+                capture_response_headers=True,
+                capture_response_json_body=True,
+            )
+            response = client.post('https://example.org/', json={'hello': 'world'})
+            checker(response)
+            assert response.json() == {'good': 'response'}
+            assert response.read() == b'{"good": "response"}'
+
+    assert exporter.exported_spans_as_dict() == snapshot(
+        [
+            {
+                'name': 'POST',
+                'context': {'trace_id': 1, 'span_id': 3, 'is_remote': False},
+                'parent': {'trace_id': 1, 'span_id': 1, 'is_remote': False},
+                'start_time': 2000000000,
+                'end_time': 3000000000,
+                'attributes': {
+                    'http.method': 'POST',
+                    'http.request.method': 'POST',
+                    'http.url': 'https://example.org/',
+                    'url.full': 'https://example.org/',
+                    'http.host': 'example.org',
+                    'server.address': 'example.org',
+                    'network.peer.address': 'example.org',
+                    'logfire.span_type': 'span',
+                    'logfire.msg': 'POST /',
+                    'http.request.header.host': ('example.org',),
+                    'http.request.header.accept': ('*/*',),
+                    'http.request.header.accept-encoding': ('gzip, deflate',),
+                    'http.request.header.connection': ('keep-alive',),
+                    'http.request.header.user-agent': (IsStr(),),
+                    'http.request.header.content-length': (IsStr(),),
+                    'http.request.header.content-type': ('application/json',),
+                    'logfire.json_schema': '{"type":"object","properties":{"http.request.body.json":{"type":"object"}}}',
+                    'http.request.body.json': '{"hello":"world"}',
+                    'http.status_code': 200,
+                    'http.response.status_code': 200,
+                    'http.flavor': '1.1',
+                    'network.protocol.version': '1.1',
+                    'http.response.header.host': ('example.org',),
+                    'http.response.header.accept': ('*/*',),
+                    'http.response.header.accept-encoding': ('gzip, deflate',),
+                    'http.response.header.connection': ('keep-alive',),
+                    'http.response.header.user-agent': (IsStr(),),
+                    'http.response.header.content-length': (IsStr(),),
+                    'http.response.header.content-type': ('application/json',),
+                    'http.response.header.traceparent': ('00-00000000000000000000000000000001-0000000000000003-01',),
```

Comment:
Isn't it a bit weird that you have both the information about the request headers, body and response headers in a single span, but the response body on another one? 🤔

Reply:
By the time the body is read, the main span is already closed, which is why a new span is needed.

The best we can do for now is to copy the attributes from the main span into the new span so that the new span has everything. But we can't put the body in the main span.